featurebyte/query_graph/node/generic.py

```diff
@@ -1227,6 +1436,61 @@ def _derive_sdk_code(
         statements.append((var_name, expression))
         return statements, var_name
 
+    def convert_to_column_remapped_node(
+        self: NodeT,
+        input_node_hashes: List[str],
+        input_node_column_remaps: List[Dict[str, str]],
+    ) -> Tuple[NodeT, Dict[str, str]]:
+        # pylint: disable=too-many-locals
+        remapped_node = self.clone()
+        column_name_remap = {}
+        input_nodes_hash = hash_input_node_hashes(input_node_hashes)
+        assert isinstance(self.parameters, JoinNodeParameters)
+        assert isinstance(remapped_node.parameters, JoinNodeParameters)
+
+        # remap left & right on columns
+        remapped_left_input_columns = []
+        remapped_left_output_columns = []
+        remapped_right_input_columns = []
+        remapped_right_output_columns = []
+        input_node_column_remap = input_node_column_remaps[0]
+        for left_in_col, left_out_cols in zip(
+            self.parameters.left_input_columns, self.parameters.left_output_columns
+        ):
+            remapped_left_in_col = input_node_column_remap.get(left_in_col, left_in_col)
+            remapped_left_input_columns.append(InColumnStr(remapped_left_in_col))
+            remapped_left_out_col = f"left_{input_nodes_hash}_{remapped_left_in_col}"
+            remapped_left_output_columns.append(OutColumnStr(remapped_left_out_col))
+            column_name_remap[str(left_out_cols)] = remapped_left_out_col
+
+        for right_in_col, right_out_cols in zip(
+            self.parameters.right_input_columns, self.parameters.right_output_columns
+        ):
+            remapped_right_in_col = input_node_column_remap.get(right_in_col, right_in_col)
+            remapped_right_input_columns.append(InColumnStr(remapped_right_in_col))
+            remapped_right_out_col = f"right_{input_nodes_hash}_{right_in_col}"
+            remapped_right_output_columns.append(OutColumnStr(remapped_right_out_col))
+            column_name_remap[str(right_out_cols)] = remapped_right_out_col
+
+        # remap the left & right columns
+        left_in_cols, left_out_cols = sort_lists_by_first_list(  # type: ignore
+            remapped_left_input_columns,
+            remapped_left_output_columns,
+        )
+        right_in_cols, right_out_cols = sort_lists_by_first_list(  # type: ignore
+            remapped_right_input_columns,
+            remapped_right_output_columns,
+        )
+        remapped_node.parameters.left_input_columns = left_in_cols
+        remapped_node.parameters.left_output_columns = left_out_cols  # type: ignore
+        remapped_node.parameters.right_input_columns = right_in_cols
+        remapped_node.parameters.right_output_columns = right_out_cols  # type: ignore
+
+        # reset metadata
+        remapped_node.parameters.metadata = None
+        remapped_node.parameters.scd_parameters = None
```

Comment:
Is it safe to reset the scd_parameters here?

Reply:
It is possible to construct the same graph structure with this `scd_parameters` set differently? If yes, we should not reset these parameters.